addons/sale_management/controllers/portal.py

```diff
@@ -55,20 +55,19 @@ def update_line_dict(self, line_id, remove=False, unlink=False, order_id=None, a
 
         if unlink or quantity <= 0:
             order_line.unlink()
-            results = self._get_portal_order_details(order_sudo)
-            results.update({
-                'unlink': True,
-                'sale_template': request.env['ir.ui.view']._render_template('sale.sale_order_portal_content', {
-                    'sale_order': order_sudo,
-                    'report_type': "html"
-                }),
-            })
-            return results
-
-        order_line.write({'product_uom_qty': quantity})
-        results = self._get_portal_order_details(order_sudo, order_line)
-
-        return results
+        else:
+            order_line.write({'product_uom_qty': quantity})
+            order_line.with_context(is_qty_changed=True)._onchange_discount()
```

Comment:
Calling an onchange on a controller is not a good practice. 

Adding a context key is not a good idea too, this will surely lead to unwanted behaviors. 

We should convert the onchange method into a compute method (probably split into 2 methods, with one if the product_uom_qty is modified), and the call to onchange won't be necessary anymore, and this will be called implicitly.

Thanks.